sql: TestDropColumnAfterMutations failed

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3499270&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3499270&tab=artifacts#/) on master @ [ecaa4920192549af1418923648769ede3de1c9c8](https://github.com/cockroachdb/cockroach/commits/ecaa4920192549af1418923648769ede3de1c9c8):


```
=== RUN   TestDropColumnAfterMutations
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations518705626
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDropColumnAfterMutations
    schema_changer_test.go:7362: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations518705626
--- FAIL: TestDropColumnAfterMutations (1.93s)
=== RUN   TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail
    schema_changer_test.go:7350: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) AND application_name='TestDropColumnAfterMutations'': expected:
        2
        
        got:
        3
        
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (0.53s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3501328&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3501328&tab=artifacts#/) on master @ [37d9f72a28a8e009bc6ba3c4d3c5780af554b92d](https://github.com/cockroachdb/cockroach/commits/37d9f72a28a8e009bc6ba3c4d3c5780af554b92d):


```
=== RUN   TestDropColumnAfterMutations
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations588100384
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDropColumnAfterMutations
    schema_changer_test.go:7362: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations588100384
--- FAIL: TestDropColumnAfterMutations (2.09s)
=== RUN   TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail
    schema_changer_test.go:7350: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) AND application_name='TestDropColumnAfterMutations'': expected:
        2
        
        got:
        3
        
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (0.66s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3508061&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3508061&tab=artifacts#/) on master @ [d62a975591ff372daae2f1be56a03109539828b4](https://github.com/cockroachdb/cockroach/commits/d62a975591ff372daae2f1be56a03109539828b4):


```
=== RUN   TestDropColumnAfterMutations
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations315615025
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDropColumnAfterMutations
    schema_changer_test.go:7362: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations315615025
--- FAIL: TestDropColumnAfterMutations (1.99s)
=== RUN   TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail
    schema_changer_test.go:7350: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) AND application_name='TestDropColumnAfterMutations'': expected:
        2
        
        got:
        3
        
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (0.61s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -7201,7 +7201,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn2.DB.ExecContext(context.Background(),
 				`
-	   SET application_name='TestDropColumnAfterMutations';
+	   SET application_name='failed-concurrent-drop-mutations';
 	   BEGIN;
 	   ALTER TABLE t ALTER COLUMN j SET NOT NULL;
 	   ALTER TABLE t ADD COLUMN k INT8 DEFAULT 42 NOT NULL UNIQUE;
@@ -7216,7 +7216,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn1.DB.ExecContext(context.Background(),
 				`
-	   SET application_name='TestDropColumnAfterMutations';
+	   SET application_name='failed-concurrent-drop-mutations';
 	   SET sql_safe_updates = false;
 	   BEGIN;
 	   ALTER TABLE t DROP COLUMN j;
@@ -7254,7 +7254,7 @@ COMMIT;
 		// Both jobs should be stuck in COMMIT, waiting for jobs to complete.
 		conn1.CheckQueryResults(t,
 			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) "+
-				"AND application_name='TestDropColumnAfterMutations'",
+				"AND application_name='failed-concurrent-drop-mutations'",
 			[][]string{{"2"}},
 		)
 
@@ -7284,7 +7284,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn2.DB.ExecContext(context.Background(),
 				`
-	   SET application_name='TestDropColumnAfterMutations';
+	   SET application_name='concurrent-drop-mutations-insert-fail';
 	   BEGIN;
 	   ALTER TABLE t ALTER COLUMN j SET NOT NULL;
 	   ALTER TABLE t ADD COLUMN k INT8 DEFAULT 42 NOT NULL;
@@ -7297,7 +7297,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn1.DB.ExecContext(context.Background(),
 				`
-	   SET application_name='TestDropColumnAfterMutations';
+	   SET application_name='concurrent-drop-mutations-insert-fail';
 	   SET sql_safe_updates = false;
 	   BEGIN;
 	   ALTER TABLE t DROP COLUMN j;
@@ -7349,7 +7349,7 @@ COMMIT;
 		// Both jobs should be stuck in COMMIT, waiting for jobs to complete.
 		conn1.CheckQueryResults(t,
 			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) "+
-				"AND application_name='TestDropColumnAfterMutations'",
+				"AND application_name='concurrent-drop-mutations-insert-fail'",
 			[][]string{{"2"}},
 		)
 

